packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js

```diff
@@ -85,10 +89,10 @@ function useResolveEditedEntityAndContext( { postId, postType } ) {
 					}
 				}
 
-				// If no template is assigned, use the default template.
-				return getDefaultTemplateId( {
-					slug: `${ postTypeToResolve }-${ editedEntity?.slug }`,
-				} );
+				const path = getPathAndQueryString( editedEntity.link );
+
+				// If no template is assigned, use the link to fetch the right template.
+				return __experimentalGetTemplateForLink( path )?.id;
```

Comment:
In my testing, this works well, i.e., the front page template is loaded in preference to the static home page in the reader settings.

However, I noticed that when opening the single page (Design > Pages > Static home page), the front page template is also loaded.

When on a page details view, should the page's template rather be loaded? The page's details in the left sidebar describe the page's meta but display's the front page template, which appears unrelated.

### Trunk

On the static page's home details view, we get the pages template, e.g., `twentytwentyfour//page`. 

<img width="528" alt="Screenshot 2023-11-21 at 12 22 42 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/97175c12-ba4f-499e-b6cb-66dcf592323a">

### This PR
Here we're loading `twentytwentyfour//front-page`

<img width="530" alt="Screenshot 2023-11-21 at 12 22 50 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/75d167e5-4271-4c82-b87f-9d6e7b30dd90">

What's the preferred way to handle page details? Should we still load the page's true template so folks can edit it as per other pages?

Reply:
I think the "front page" should load even in that case because that's the actual template that is used for that specific page anyway in the frontend.

Reply:
That's correct, Front Page will be used for the site homepage regardless.

Reply:
Thanks for the confirmation. 👍🏻 

I suppose I just found it confusing to see the page details on the left, including excerpt etc, and not the corresponding representation of that page on the right.